### PR TITLE
Goraexplorer needed changes

### PR DESCRIPTION
**Pull request by alfonsonishikawa**
This set of commits implements the following functionalities:

- Compile schemas from Strings and not only from files.
- Allow to configure the mapping of a HBase backed datastore when creating the datastore
- Added a new factory method to create a datastore using class' names and not depending on generics 
- Updated Persistent to get the related schema.
- Added #clone in Persistent
- Enabled the possibility to retrieve information from the native backend behind a datastore. This information is used in the [schema's wizard of GoraExplorer](https://bitbucket.org/alfonsonishikawa/goraexplorer/wiki/User%20manual#markdown-header-handling-schemas-and-mappings-with-the-wizard).
- Added gora-pig module to access Gora from Pig.

[More details on gora-pig](http://people.apache.org/~alfonsonishikawa/gora-109.html)

Please be picky on this pull request. Any doubt, anything, ask for it.

### CODE REVIEW

File: bin/compile-examples.sh [hunk 1]
```diff
@@ -36,8 +36,57 @@ DIR=$GORA_HOME/$MODULE/src/examples/avro/
 OUTDIR=$GORA_HOME/$MODULE/src/examples/java
 GORA_BIN=$GORA_HOME/bin/gora
 
+rm -r $OUTDIR/org/apache/gora/examples/generated/*
```
Comment (alfonsonishikawa) [inline bin/compile-examples.sh]: I found that the compile-examples script was vastly outdated.  I updated it, but still needs to have compiled the module 'gora-compiler-cli', so, running the first time must do:

```
mvn package
// wait for compile crash after gora-compiler-cli module
bin/compile-examples.sh
mvn package
```

File: gora-compiler/src/main/java/org/apache/gora/compiler/GoraCompiler.java [hunk 2]
```diff
@@ -86,6 +88,31 @@ public static void compileSchema(File[] srcFiles, File dest, LicenseHeaders lice
     }
   }
 
+  /**
```
Comment (alfonsonishikawa) [inline gora-compiler/src/main/java/org/apache/gora/compiler/GoraCompiler.java]: Schema compilation from a String (instead of a file), with a configurable templatesPath.
I actually used it as `GoraCompiler.compileSchema(avroSchemaDefinition, outputDirectory, null)`

File: gora-compiler/src/main/velocity/org/apache/gora/compiler/templates/record.vm [hunk 3]
```diff
@@ -180,6 +180,11 @@ public class ${this.mangle($schema.getName())}#if ($schema.isError()) extends or
     return new #if ($schema.getNamespace())$schema.getNamespace().#end${this.mangle($schema.getName())}.Builder(other);
   }
   
+  @Override
```
Comment (alfonsonishikawa) [inline gora-compiler/src/main/velocity/org/apache/gora/compiler/templates/record.vm]: A clone method. I used it [here](https://bitbucket.org/alfonsonishikawa/goraexplorer/src/12f226408e7f9686aa6b5975692edf270b94182e/src/main/java/com/nishilua/goraexplorer/tables/TableEntityServiceImpl.java#lines-118) because `get()` reuses the instance.

File: gora-core/src/examples/avro/employee.json [hunk 4]
```diff
@@ -14,20 +14,22 @@
           "name": "WebPage",
```
Comment (alfonsonishikawa) [inline gora-core/src/examples/avro/employee.json]: Updated the outdated schema.

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java [hunk 5]
```diff
@@ -80,15 +81,20 @@
  * DataStore for HBase. Thread safe.
  *
  */
-public class HBaseStore<K, T extends PersistentBase> extends DataStoreBase<K, T>
```
Comment (alfonsonishikawa) [inline gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java]: I don't remember why this Configurable was deleted. Is it needed?

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java [hunk 6]
```diff
@@ -80,15 +81,20 @@
...
 
   public static final String PARSE_MAPPING_FILE_KEY = "gora.hbase.mapping.file";
 
   public static final String DEFAULT_MAPPING_FILE = "gora-hbase-mapping.xml";
 
+  /**
+   * Key at DataStore Properties (same as gora.properties) to hold a mapping loaded from memory, instead from filesystem like
+   * the one at PARSE_MAPPING_FILE_KEY. If this key is present, the mapping is loaded from the value instead of gora-hbase-mapping.xml
+   */
+  public static final String XML_MAPPING_DEFINITION = "gora.mapping" ;
```
Comment (alfonsonishikawa) [inline gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java]: When creating the DataStore (for HBase!) the Properties can have a key `gora.mapping` with the XML document of the mapping. If it exists, this mapping is loaded instead of the file one at `gora-hbase-mapping.xml`.

File: pom.xml [hunk 7]
```diff
@@ -757,9 +758,9 @@
     <module>gora-maven-plugin</module>
     <module>gora-mongodb</module>
     <module>gora-solr</module>
+    <module>gora-aerospike</module>
```
Comment (alfonsonishikawa) [inline pom.xml]: gora-aerospike must be before gora-tutorial and sources-dist to be able to compile successfully for the first time. Dependency issue fixed.

File: gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled [hunk 8]
```diff
@@ -0,0 +1,352 @@
+package org.apache.gora.pig;
```
Comment (alfonsonishikawa) [inline gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled]: Gora-pig does not have JUnit tests!
I have been struggling to launch a Hadoop+HBase cluster. It seems with Hadoop 1 + HBase 1.0.x (or alike) I got it working in Hadoop 1 with an old Gora version: JUnit started both pseudo distributed and did the job. But with HBase 1.2.6 `utility.startMiniCluster();` and `utility.startMiniMapReduceCluster();` tries to launch a Hadoop 1 cluster. And it crashes too.

At the end, I developed and tested gora-pig directly with HDP 2.6.

**A hand is needed to make this run** :\ Thank you.

File: bin/compile-examples.sh (hunk 1 above)
Comment (lewismc) [reply bin/compile-examples.sh]: Yep OK

File: gora-compiler/src/main/java/org/apache/gora/compiler/GoraCompiler.java (hunk 2 above)
Comment (lewismc) [reply gora-compiler/src/main/java/org/apache/gora/compiler/GoraCompiler.java]: I like this

File: gora-compiler/src/main/velocity/org/apache/gora/compiler/templates/record.vm (hunk 3 above)
Comment (lewismc) [reply gora-compiler/src/main/velocity/org/apache/gora/compiler/templates/record.vm]: You should have referenced line 121 not 118.

File: gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java [hunk 9]
```diff
@@ -1,20 +0,0 @@
-/**
```
Comment (lewismc) [inline gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java]: Why remove?

File: gora-core/src/main/java/org/apache/gora/persistency/Persistent.java [hunk 10]
```diff
@@ -113,4 +118,13 @@
...
    */
   Persistent newInstance();
+  
+  /**
+   * Returns the avro's data schema
+   * @return the parsed schema definition
+   */
+  public Schema getSchema();
+  
+  public Persistent clone() throws CloneNotSupportedException;
```
Comment (lewismc) [inline gora-core/src/main/java/org/apache/gora/persistency/Persistent.java]: Can you provide Javadoc for this? It is a primary method in a very important interface.

File: gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java [hunk 11]
```diff
@@ -251,6 +254,70 @@ D getDataStore( Class<D> dataStoreClass, Class<K> keyClass,
     }
   }
 
+  /**
+   * Instantiate a new {@link DataStore}. Uses default properties. Uses 'null' schema.
+   * 
+   * @param dataStoreClass The datastore implementation class <i>as string</i>.
```
Comment (lewismc) [inline gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java]: Not required this is not a parameter

File: gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java [hunk 12]
```diff
@@ -412,7 +484,7 @@ public static String getMappingFile(Properties properties, DataStore<?,?> store
     return mappingFilename;
   }
 
-  private static String getDefaultDataStore(Properties properties) {
+  public static String getDefaultDataStore(Properties properties) {
```
Comment (lewismc) [inline gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java]: Why the change here?

File: gora-core/src/main/java/org/apache/gora/store/DataStoreMetadataFactory.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  }
+  
+  /**
+   * Creates a metadata analyzer given the Hadoop Configuration instance and the gora properties to use. It uses this properties
+   * to infer the metadata analyzer to create.
+   * @throws GoraException
+   * @throws ClassNotFoundException - Exception thrown when does not exists a suitable metadata analyzer class
+   */
+  public static DataStoreMetadataAnalyzer createAnalyzer(Configuration configuration, Properties properties) throws GoraException, ClassNotFoundException {    
+    String metadataAnalyzerClassName = DataStoreFactory.getDefaultDataStore(properties) + "MetadataAnalyzer";
```
Comment (lewismc) [inline gora-core/src/main/java/org/apache/gora/store/DataStoreMetadataFactory.java]: Aha, I now see why you want this to be made public.

File: gora-goraci/src/main/java/org/apache/gora/goraci/generated/package-info.java [hunk 14]
```diff
@@ -1,21 +0,0 @@
-/**
```
Comment (lewismc) [inline gora-goraci/src/main/java/org/apache/gora/goraci/generated/package-info.java]: Why remove?

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java (hunk 5 above)
Comment (lewismc) [reply gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java]: It wasn't just for HBaseStore?

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java [hunk 15]
```diff
@@ -734,14 +756,13 @@ private void setField(T persistent, Field field, List list) {
   }
 
   @SuppressWarnings("unchecked")
-  private HBaseMapping readMapping(String filename) throws IOException {
+  private HBaseMapping readMapping(InputStream mappingStream) throws IOException {
```
Comment (lewismc) [inline gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java]: I like this. The quicker work with ByteStream the better.

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStoreMetadataAnalyzer.java [hunk 16]
```diff
@@ -0,0 +1,75 @@
+package org.apache.gora.hbase.store;
```
Comment (lewismc) [inline gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStoreMetadataAnalyzer.java]: License header please

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseTableMetadata.java [hunk 17]
```diff
@@ -0,0 +1,27 @@
+package org.apache.gora.hbase.store;
```
Comment (lewismc) [inline gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseTableMetadata.java]: License header please.

File: gora-pig/pom.xml [hunk 18]
```diff
@@ -0,0 +1,180 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lewismc) [inline gora-pig/pom.xml]: License header.

File: gora-pig/pom.xml [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+        </dependency>
+
+        <!-- Gora Internal Dependencies -->
+        <dependency>
+            <groupId>org.apache.gora</groupId>
+            <artifactId>gora-core</artifactId>
+            <scope>compile</scope>
+        </dependency>
+
+<!--         <dependency> -->
```
Comment (lewismc) [inline gora-pig/pom.xml]: Just remove commented out dependencies.

File: gora-pig/src/main/java/org/apache/gora/pig/GoraDeleteStorage.java [hunk 20]
```diff
@@ -0,0 +1,111 @@
+package org.apache.gora.pig;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/GoraDeleteStorage.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/GoraDeleteStorage.java [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ * Storage to delete rows by key.
+ * 
+ * Given a relation with schema (key:chararray) rows, the following will delete
+ * all rows with that keys:
+ *                          
+ * <pre>
+ * STORE webpages INTO '.' USING org.apache.gora.pig.GoraDeleteStorage('{
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/GoraDeleteStorage.java]: This is killer. So cool.

File: gora-pig/src/main/java/org/apache/gora/pig/ResourceFieldSchemaWithIndex.java [hunk 22]
```diff
@@ -0,0 +1,31 @@
+package org.apache.gora.pig;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/ResourceFieldSchemaWithIndex.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/StorageConfiguration.java [hunk 23]
```diff
@@ -0,0 +1,140 @@
+package org.apache.gora.pig;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/StorageConfiguration.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraInputFormat.java [hunk 24]
```diff
@@ -0,0 +1,122 @@
+package org.apache.gora.pig.mapreduce;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraInputFormat.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java [hunk 25]
```diff
@@ -0,0 +1,86 @@
+package org.apache.gora.pig.mapreduce;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java [hunk 26]
```diff
@@ -0,0 +1,86 @@
...
+          ) ;      
+    }
+    
+    return new GoraRecordWriter(this.dataStore, context);
+  }
+
+
+  @Override
+  public void checkOutputSpecs(JobContext context) throws IOException, InterruptedException {
+    // Nothing
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java]: throw unsupported exception?

File: gora-pig/src/main/java/org/apache/gora/pig/udf/MapDeleteElements.java [hunk 27]
```diff
@@ -0,0 +1,67 @@
+package org.apache.gora.pig.udf;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/udf/MapDeleteElements.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/udf/MapMerge.java [hunk 28]
```diff
@@ -0,0 +1,33 @@
+package org.apache.gora.pig.udf;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/udf/MapMerge.java]: License header.

File: gora-pig/src/main/java/org/apache/gora/pig/util/PersistentUtils.java [hunk 29]
```diff
@@ -0,0 +1,147 @@
+package org.apache.gora.pig.util;
```
Comment (lewismc) [inline gora-pig/src/main/java/org/apache/gora/pig/util/PersistentUtils.java]: License header.

File: gora-pig/src/test/examples/README.txt [hunk 30]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (lewismc) [inline gora-pig/src/test/examples/README.txt]: Can you make this an .md file?

File: gora-pig/src/test/examples/example.pig [hunk 31]
```diff
@@ -0,0 +1,4 @@
+set job.name 'GoraStorage test' ;
```
Comment (lewismc) [inline gora-pig/src/test/examples/example.pig]: License header if possible.

File: gora-pig/src/test/examples/example.pig [hunk 32]
```diff
@@ -0,0 +1,4 @@
...
+register gora/*.jar ;
+webpage = LOAD '.' using org.apache.gora.pig.GoraStorage('java.lang.String','admin.WebPage','baseUrl,status,content') ;
+dump webpage;
```
Comment (lewismc) [inline gora-pig/src/test/examples/example.pig]: We cannot store webpage.jar in the Gora source... we need some way of building this via compile examples maybe?

File: gora-pig/src/test/resources/gora.properties [hunk 33]
```diff
@@ -0,0 +1,4 @@
+gora.datastore.default=org.apache.gora.hbase.store.HBaseStore
```
Comment (lewismc) [inline gora-pig/src/test/resources/gora.properties]: License header.

File: pom.xml (hunk 7 above)
Comment (lewismc) [reply pom.xml]: Good catch.

File: pom.xml [hunk 34]
```diff
@@ -832,6 +833,12 @@
     <checksum-maven-plugin.version>1.0.1</checksum-maven-plugin.version>
     <maven-clean-plugin.version>2.5</maven-clean-plugin.version>
 
+    <!-- Pig Dependencies -->
+    <pig.version>0.16.0</pig.version>
+    <automaton.version>1.11-8</automaton.version>
```
Comment (lewismc) [inline pom.xml]: Is this required? What do you use it for?

File: pom.xml [hunk 35]
```diff
@@ -1546,6 +1553,40 @@
...
+
+      <dependency>
+          <groupId>org.antlr</groupId>
+          <artifactId>antlr-runtime</artifactId>
+          <version>${antlr.version}</version>
+          <scope>test</scope>
+      </dependency>
+
+      <dependency>
+          <groupId>com.linkedin.datafu</groupId>
```
Comment (lewismc) [inline pom.xml]: Can you not use org.apache.datafu artifact?

File: gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java (hunk 5 above)
Comment (alfonsonishikawa) [reply gora-hbase/src/main/java/org/apache/gora/hbase/store/HBaseStore.java]: Yes, it seems it was only for HBaseStore, from the initial import of the source code 8 years ago.

File: gora-pig/pom.xml (hunk 19 above)
Comment (alfonsonishikawa) [reply gora-pig/pom.xml]: They are commented because they were the ones used for -the not working- JUnit tests. Anyway, they will get rotten, so you are right :) I will delete them.

File: gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java (hunk 26 above)
Comment (alfonsonishikawa) [reply gora-pig/src/main/java/org/apache/gora/pig/mapreduce/PigGoraOutputFormat.java]: Implementing an mandatory abstract method. We just don't check and take it as free of errors. The output will fail always if an IOException/InterrupedException is thrown.

File: gora-pig/src/test/examples/example.pig (hunk 32 above)
Comment (alfonsonishikawa) [reply gora-pig/src/test/examples/example.pig]: This is actually an usage example. It could be completely deleted and commented on the README.md how to create the file. Or even delete the folder completely.

**Comment by djkevincr** [comment]
@alfonsonishikawa Are you done with updating PR as per the reviews? Is there anything else that we should address or do you need further time on this? Its awesome if we can get to the release coming shortly?

File: gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java (hunk 9 above)
Comment (alfonsonishikawa) [reply gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java]: It seems that compile-examples.sh just deletes generated/* to regenerate the classes. I will update bin/compile-examples.sh to just delete .java files

**Comment by alfonsonishikawa** [comment]
Hi! I am updating this pull request with the fixes from your comments. Will finish it this week. Thanks!

File: gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java (hunk 12 above)
Comment (alfonsonishikawa) [reply gora-core/src/main/java/org/apache/gora/store/DataStoreFactory.java]: Sadly the MetadataAnalyzer (to retrieve native information from the backend) depends on the name of the DataStore.
I acknowledge I went into a rush about implementing the feature, so any suggestions about are welcomed :)
I see you found out why by yourself! (https://github.com/apache/gora/pull/135/files/2974126a934bddc526830182b402767c0f1eefd0?file-filters%5B%5D=.java&hide-deleted-files=true#diff-d9701987f6844457a897d30a5786e5a4) :D

File: conf/log4j.properties [hunk 36]
```diff
@@ -24,7 +24,7 @@ log4j.threshhold=ALL
 log4j.rootLogger=${gora.root.logger}
 
 #special logging requirements for some commandline tools
-log4j.logger.org.apache.gora.compiler.GoraCompiler=INFO,console
+log4j.logger.org.apache.gora.compiler.GoraCompiler=INFO
```
Comment (alfonsonishikawa) [inline conf/log4j.properties]: Logs in console about compiled files appeared twice because the root,logger is already configured to log to console.

File: gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java (hunk 9 above)
Comment (alfonsonishikawa) [reply gora-core/src/examples/java/org/apache/gora/examples/generated/package-info.java]: Now the package-info.java files are moved to the parent directory temporarily at `compile-examples.sh` before deleting the generated classes.

File: gora-pig/src/test/examples/example.pig (hunk 32 above)
Comment (alfonsonishikawa) [reply gora-pig/src/test/examples/example.pig]: Deleted `webpage.jar` and updated README.md to describe the steps to execute an example.

File: gora-pig/src/test/examples/README.md [hunk 37]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (alfonsonishikawa) [inline gora-pig/src/test/examples/README.md]: The Gora-Pig example README.txt renamed to README.md

**Comment by alfonsonishikawa** [comment]
Hi! Finished the changes. You can see all in the [Files Tab](https://github.com/apache/gora/pull/135/files?file-filters%5B%5D=).

Thank you!!

**Comment by djkevincr** [comment]
Locally tested the  PR, build passes without any test failures.

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Gora ........................................ SUCCESS [  2.381 s]
[INFO] Apache Gora :: Compiler ............................ SUCCESS [  3.077 s]
[INFO] Apache Gora :: Compiler-CLI ........................ SUCCESS [  1.385 s]
[INFO] Apache Gora :: Core ................................ SUCCESS [01:56 min]
[INFO] Apache Gora :: Pig ................................. SUCCESS [  3.235 s]
[INFO] Apache Gora :: Accumulo ............................ SUCCESS [08:07 min]
[INFO] Apache Gora :: HBase ............................... SUCCESS [03:24 min]
[INFO] Apache Gora :: Cassandra - CQL ..................... SUCCESS [01:53 min]
[INFO] Apache Gora :: GoraCI .............................. SUCCESS [  3.998 s]
[INFO] Apache Gora :: Infinispan .......................... SUCCESS [01:22 min]
[INFO] Apache Gora :: JCache .............................. SUCCESS [01:21 min]
[INFO] Apache Gora :: OrientDB ............................ SUCCESS [01:48 min]
[INFO] Apache Gora :: Dynamodb ............................ SUCCESS [  4.441 s]
[INFO] Apache Gora :: CouchDB ............................. SUCCESS [  4.872 s]
[INFO] Apache Gora :: Maven Plugin ........................ SUCCESS [  3.021 s]
[INFO] Apache Gora :: MongoDB ............................. SUCCESS [02:04 min]
[INFO] Apache Gora :: Solr ................................ SUCCESS [02:59 min]
[INFO] Apache Gora :: Aerospike ........................... SUCCESS [  2.849 s]
[INFO] Apache Gora :: Ignite .............................. SUCCESS [02:56 min]
[INFO] Apache Gora :: Tutorial ............................ SUCCESS [  6.486 s]
[INFO] Apache Gora :: Sources-Dist ........................ SUCCESS [  0.364 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28:32 min
[INFO] Finished at: 2019-03-18T23:14:41+05:30
[INFO] Final Memory: 101M/1679M
[INFO] ------------------------------------------------------------------------

File: gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled (hunk 8 above)
Comment (djkevincr) [reply gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled]: Were you able to try the same with HBase 2 upgrade?

**Comment by djkevincr** [comment]
@alfonsonishikawa One concern I do have is, I noticed record.vm velocity template changes, hopefully I think you have regenerated all the AVRO databean classes again to avoid any inconsistent updates, due to multiple updates to velocity template.
This is really great :) as first step. We can continue this work, with the improvements you suggested to me offline. @lewismc Do you have any concern over your review on this PR?

File: gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled (hunk 8 above)
Comment (alfonsonishikawa) [reply gora-pig/src/test/java/org/apache/gora/pig/GoraStorageTest.java-disabled]: > Were you able to try the same with HBase 2 upgrade?

Hi! No. I will try, though. Thanks!

**Comment by lewismc** [comment]
> Do you have any concern over your review on this PR?

No

**Comment by alfonsonishikawa** [comment]
Oh! Thank you for the pull!

I was working on the Gora-Pig tests as @djkevincr suggested. I will continue working on them anyway but will open a Jira issue.
Thank you both for the reviews!

**Comment by djkevincr** [comment]
@alfonsonishikawa I have merged you PR, thank you very much for your hard work on this :) I went through your design documents and all looked very impressive. It s really nice to see Apache Pig integrated to our project. Let's continue this work :)

**Comment by djkevincr** [comment]
@alfonsonishikawa Lets create a new ticket and address separately :)
